great_expectations/data_context/types/resource_identifiers.py

```diff
@@ -253,7 +253,10 @@ def from_fixed_length_tuple(cls, tuple_):
         return cls.from_tuple(tuple_)
 
     def __repr__(self):
-        return f"{self.__class__.__name__}::{self.resource_type}::{self.ge_cloud_id}"
+        repr = f"{self.__class__.__name__}::{self.resource_type}::{self.ge_cloud_id}"
+        if self.resource_name:
+            repr += f"::{self.resource_name}"
+        return repr
```

Comment:
If this is already a well-established pattern then ignore this but...

One suggestion I've heard in the past regarding how to define a `__repr__` is to make it possible to re-create the object using the `repr()` output.

So for example...
```python
class Car:
    def __init__(self, color, mileage):
        self.color = color
        self.mileage = mileage

    def __repr__(self):
       return (f'{self.__class__.__name__}('
               f'{self.color!r}, {self.mileage!r})')

    def __str__(self):
        return f'a {self.color} car'

my_car = Car('blue', 500)
print(repr(my_car))
print(str(my_car))

... Car('blue', 500)
... a blue car
```


https://dbader.org/blog/python-repr-vs-str

Reply:
ooh I like that! I think we'll probably want to keep the reprs as-is for now due to the fact that we don't consistently do this